src/vmm/src/builder.rs

```diff
@@ -496,27 +496,20 @@ pub fn build_microvm_from_snapshot(
     }
 
     // Restore vcpus kvm state.
-    #[cfg(target_arch = "aarch64")]
-    {
-        for (vcpu, state) in vcpus.iter_mut().zip(microvm_state.vcpu_states.iter()) {
-            vcpu.kvm_vcpu
-                .restore_state(vmm.vm.fd(), state)
-                .map_err(VcpuError::VcpuResponse)
-                .map_err(BuildMicrovmFromSnapshotError::RestoreVcpus)?;
-        }
-        let mpidrs = construct_kvm_mpidrs(&microvm_state.vcpu_states);
-        // Restore kvm vm state.
-        vmm.vm.restore_state(&mpidrs, &microvm_state.vm_state)?;
-    }
-
-    #[cfg(target_arch = "x86_64")]
     for (vcpu, state) in vcpus.iter_mut().zip(microvm_state.vcpu_states.iter()) {
         vcpu.kvm_vcpu
             .restore_state(state)
```

Comment:
Nice! The inconsistency in signatures here has been bothering me for a while haha